stl/inc/functional

```diff
@@ -1299,7 +1299,11 @@ inline void __stdcall _Function_deallocate_large_default_aligned(_Move_only_func
 
 template <size_t _Align>
 void __stdcall _Function_deallocate_large_overaligned(_Move_only_function_data& _Self) noexcept {
+#ifdef __cpp_aligned_new
     ::operator delete (_Self._Large_fn_ptr<void>(), align_val_t{_Align});
+#else // ^^^ __cpp_aligned_new / !__cpp_aligned_new vvv
+    _CSTD _aligned_free(_Self._Large_fn_ptr<void>());

```

Comment:
I'm concerned that the treatment of overaligned `move_only_function` targets in `/Zc:alignedNew-` mode here is inconsistent with how we treat overaligned types everywhere else in the STL. Admittedly we couldn't change the handling for things that existed in C++14 without breaking existing code and `move_only_function` is new, but I'm concerned that `/Zc:alignedNew-` users will _want_ the C++14 behavior and will report this inconsistency as a bug.

Am I being overly paranoid? Am I just wrong? Is there some urgent reason to handle `alignedNew-` differently here?

Reply:
I think the existing handling for `/Zc:alignedNew-` is broken, as it may allocate improperly aligned storage. I believe that creating an object at an improperly aligned address is an unreasonable request, because it seems that the Standard doesn't support it ([[basic.life]/(1.1)](https://eel.is/c++draft/basic.life#1.1))... although it's already non-Standard to use `/Zc:alignedNew-` in C++23 mode.

If user-replaced (normal) `operator new` is wanted, I think we should allocate more bytes (perhaps `sizeof(_Vt) + alignof(_Vt)`?), manually align the address, and then record the offset (or something equivalent) before the object.

Reply:
See https://developercommunity.visualstudio.com/t/10036384#T-N10037977

> The only reason I am using `/Zc:alignedNew-` at all is to eliminate the space/time overhead of overaligned allocations due to the inability to customize `__STDCPP_DEFAULT_NEW_ALIGNMENT__` while using a global operator new that provides stronger alignment guarantees.

So the user wants to specifically avoid library-provided overalignment, and they are doing overalignment for entire global new. By doing anything except global new, we defeat user's purpose.